evennia/contrib/turnbattle/tb_magic.py

```diff
@@ -0,0 +1,1285 @@
+            character's HP.
+
+    Notes:
+        By default, returns a random integer from 15 to 25 without using any
+        properties from either the attacker or defender.
+
+        Again, this can be expanded upon.
+    """
+    # For this example, just generate a number between 15 and 25.
+    damage_value = randint(15, 25)
+    return damage_value
+
+
+def apply_damage(defender, damage):
+    """
+    Applies damage to a target, reducing their HP by the damage amount to a
+    minimum of 0.
+
+    Args:
+        defender (obj): Character taking damage
+        damage (int): Amount of damage being taken
+    """
+    defender.db.hp -= damage  # Reduce defender's HP by the damage dealt.
+    # If this reduces it to 0 or less, set HP to 0.
+    if defender.db.hp <= 0:
+        defender.db.hp = 0
+
+def at_defeat(defeated):
+    """
+    Announces the defeat of a fighter in combat.
+    
+    Args:
+        defeated (obj): Fighter that's been defeated.
+    
+    Notes:
+        All this does is announce a defeat message by default, but if you
+        want anything else to happen to defeated fighters (like putting them
+        into a dying state or something similar) then this is the place to
+        do it.
+    """
+    defeated.location.msg_contents("%s has been defeated!" % defeated)
+
+def resolve_attack(attacker, defender, attack_value=None, defense_value=None):
+    """
+    Resolves an attack and outputs the result.
+
+    Args:
+        attacker (obj): Character doing the attacking
+        defender (obj): Character being attacked
+
+    Notes:
+        Even though the attack and defense values are calculated
+        extremely simply, they are separated out into their own functions
+        so that they are easier to expand upon.
+    """
+    # Get an attack roll from the attacker.
+    if not attack_value:
+        attack_value = get_attack(attacker, defender)
+    # Get a defense value from the defender.
+    if not defense_value:
+        defense_value = get_defense(attacker, defender)
+    # If the attack value is lower than the defense value, miss. Otherwise, hit.
+    if attack_value < defense_value:
+        attacker.location.msg_contents("%s's attack misses %s!" % (attacker, defender))
+    else:
+        damage_value = get_damage(attacker, defender)  # Calculate damage value.
+        # Announce damage dealt and apply damage.
+        attacker.location.msg_contents("%s hits %s for %i damage!" % (attacker, defender, damage_value))
+        apply_damage(defender, damage_value)
+        # If defender HP is reduced to 0 or less, call at_defeat.
+        if defender.db.hp <= 0:
+            at_defeat(defender)
+
+def combat_cleanup(character):
+    """
+    Cleans up all the temporary combat-related attributes on a character.
+
+    Args:
+        character (obj): Character to have their combat attributes removed
+
+    Notes:
+        Any attribute whose key begins with 'combat_' is temporary and no
```

Comment:
Attributes starting with `combat_` does not seem too unlikely. Since this is not warned for, this could cause some unfortunate bugs and weirdness if people just drop this into their existing code.  A safer (and more efficient) way if you want to use temporary Attributes like this is to store the `Attribute` with a `category` set to `combat`. It can then be named whatever you like, and you later just delete all attributes of the `combat` category: 

```
obj.attributes.add("Burning bonus", 1000, category="combat")
obj.attributes.add("Disadvantage", -100 , category="combat")
# removing all of a given category
obj.attributes.clear(category="combat") 
```

The even cleaner solution would be to store all such temporary variables on the Combat-Handler Script instead, indexed with who they belong to. That way they will automatically go away when the combat ends and there is never any risk of any leftovers.

Reply:
I had no idea attribute categories were a thing! Thats probably a better way to do it, but it may be a while before I can integrate it properly, since it'll be a big change in every module. I'll save this for another PR.